php-plugin/src/test/java/fr/greencodeinitiative/php/PhpRuleRepositoryTest.java

```diff
@@ -16,42 +16,80 @@
  */
 package fr.greencodeinitiative.php;
 
-import static org.assertj.core.api.Assertions.assertThat;
 import org.assertj.core.api.SoftAssertions;
-import org.junit.Before;
-import org.junit.Test;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.sonar.api.SonarRuntime;
 import org.sonar.api.server.rule.RulesDefinition;
+import org.sonar.api.utils.Version;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+
+class PhpRuleRepositoryTest {
 
-public class PhpRuleRepositoryTest {
+  private RulesDefinition.Repository repository;
 
-  private PhpRuleRepository phpRuleRepository;
-  private RulesDefinition.Context context;
+    @BeforeEach
+  void init() {
+    // TODO: Remove this check after Git repo split
+    if (PhpRuleRepository.class.getResource("/io/ecocode/rules/php/EC4.json") == null) {
+      String message = "'ecocode-rules-specification' resources corrupted. Please check build of 'ecocode-rules-specification' module";
+      if (System.getProperties().keySet().stream().anyMatch(k -> k.toString().startsWith("idea."))) {
+        message += "\n\nOn 'IntelliJ IDEA':" +
+                "\n1. go to settings :" +
+                "\n   > Build, Execution, Deployment > Build Tools > Maven > Runner" +
+                "\n2. check option:" +
+                "\n   > Delegate IDE build/run actions to Maven" +
+                "\n3. Click on menu: " +
+                "\n   > Build > Build Project"
+        ;
+      }
+      fail(message);
+    }
 
-  @Before
-  public void init() {
-    phpRuleRepository = new PhpRuleRepository();
-    context = new RulesDefinition.Context();
-    phpRuleRepository.define(context);
+    final SonarRuntime sonarRuntime = mock(SonarRuntime.class);
```

Comment:
@utarwyn could you check here if your idea for construction of SonarRuntime mock is ok for you, please ?

Reply:
Hello! Yes it's the right way to instanciate a fake SonarRuntime object in test files 👍